tensorboard/webapp/util/ui_selectors_test.ts

```diff
@@ -95,46 +118,130 @@ describe('ui_selectors test', () => {
+                '234/run1': buildRun({id: '234/run1', name: 'run1'}),
+                '234/run2': buildRun({id: '234/run2', name: 'run2'}),
+                '234/run3': buildRun({id: '234/run3', name: 'run3'}),
+              },
+              regexFilter: '^r.n[1]',
+            })
+          ),
+          ...buildStateFromExperimentsState(
+            buildExperimentState({
+              experimentMap: {
+                '234': buildExperiment({id: '234', name: 'Experiment 234'}),
+              },
+            })
+          ),
+        };
 
-      expect(getCurrentRouteRunSelection(state)).toEqual(
-        new Map([
-          ['run1', true],
-          ['run2', false],
-          ['run3', false],
-        ])
-      );
+        expect(getCurrentRouteRunSelection(state)).toEqual(
+          new Map([
+            ['234/run1', true],
+            ['234/run2', false],
+            ['234/run3', false],
+          ])
+        );
+      });
+
+      it('filters run name, experiment name, and alias in compare mode', () => {
+        const state = {
+          ...buildStateFromAppRoutingState(
+            buildAppRoutingState({
+              activeRoute: buildRoute({
+                routeKind: RouteKind.COMPARE_EXPERIMENT,
+                pathname: '/compare/apple:123,banana:234/',
+                params: {experimentIds: 'apple:123,banana:234'},
+              }),
+            })
+          ),
+          ...buildStateFromRunsState(
+            buildRunsState({
+              selectionState: new Map([
+                [
+                  '["123","234"]',
+                  new Map([
+                    ['123/run1', false],
+                    ['123/run2', true],
+                    ['234/run1', true],
+                    ['234/run2', true],
+                    ['234/run3', false],
+                  ]),
+                ],
+              ]),
+              runIds: {
+                '123': ['123/run1', '123/run2'],
+                '234': ['234/run1', '234/run2', '234/run3'],
+              },
+              runMetadata: {
+                '123/run1': buildRun({id: '123/run1', name: 'run1'}),
+                '123/run2': buildRun({id: '123/run2', name: 'run2'}),
+                '234/run1': buildRun({id: '234/run1', name: 'run1'}),
+                '234/run2': buildRun({id: '234/run2', name: 'run2'}),
+                '234/run3': buildRun({id: '234/run3', name: 'run3'}),
+              },
+              regexFilter: '^(apple/..3|r.n[1])',
+            })
+          ),
+          ...buildStateFromExperimentsState(
+            buildExperimentState({
+              experimentMap: {
+                '123': buildExperiment({id: '123', name: 'Experiment 123'}),
+                '234': buildExperiment({id: '234', name: 'Experiment 234'}),
+              },
+            })
+          ),
+        };
+
+        expect(getCurrentRouteRunSelection(state)).toEqual(
+          new Map([
+            ['123/run1', false],
```

Comment:
The test could use some comments to help explain the result.
I look at the regex filter '^(apple/..3|r.n[1])' and wonder two things:

* How does the apple/..3 portion impact anything? It doesn't seem like it could select anything.
* the r.n[1] portion should select both 123/run1 and 234/run1 but it only selects the latter.